### ISSUE TITLE
[Bug?] MatMenuModule throws injection error on lazy loaded modules

### ISSUE THREAD

**Issue by vlio20** [angular/material2#11273]
I have a module "A" which loads (lazy) other module "B":
Here is "A" module setting:
```
@NgModule({
  declarations: [
    AppComponent,
    ErrorComponent
  ],
  imports: [
    HttpClientModule,
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    // MatMenuModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AModule {
}
```

And here is module "B":
```
@NgModule({
  imports: [
    CommonModule,
    HttpClientModule,
    MainRoutingModule,
    MatSidenavModule,
    MatIconModule,
    MatToolbarModule,
    MatButtonModule,
    MatListModule,
    MatSnackBarModule,
    MatMenuModule // <-
  ],
  declarations: [...],
  providers: [
    ...
  ]
})
export class B {
}

```

In the above configuration I have the following error:
```
core.js:1601 ERROR Error: Uncaught (in promise): Error: StaticInjectorError(AppModule)[MatMenuTrigger -> Overlay]: 
  StaticInjectorError(Platform: core)[MatMenuTrigger -> Overlay]: 
    NullInjectorError: No provider for Overlay!
Error: StaticInjectorError(AppModule)[MatMenuTrigger -> Overlay]: 
  StaticInjectorError(Platform: core)[MatMenuTrigger -> Overlay]: 
    NullInjectorError: No provider for Overlay!
    at NullInjector.push../node_modules/@angular/core/fesm5/core.js.NullInjector.get (core.js:979)
    at resolveToken (core.js:1232)
    at tryResolveToken (core.js:1182)
    at StaticInjector.push../node_modules/@angular/core/fesm5/core.js.StaticInjector.get (core.js:1077)
```

Only after adding `MatMenuModule` to the main module "A" the error is gone. 
Is it a bug? why doesn't it happens with other components?

**Comment by crisbeto** [comment]
Closing as a duplicate of https://github.com/angular/material2/issues/10820.
